.github/workflows/e2e_test_for_releases.yml

```diff
@@ -2,19 +2,111 @@ name: AWS E2E for releases and pre-releases
 
 on:
   workflow_dispatch:
-  push:
-    tags:
-      - v*
-    paths-ignore:
-      - '**.md'
-      - 'cloudformation/*.yml'
-
-permissions:
-  id-token: write # This is required for requesting the JWT
-  contents: read  # This is required for actions/checkout
+  pull_request:
+    branches:
+      - main
+#  push:
+#    tags:
+#      - v*
 
+# todo: find a way to not use hardcoded `rc`
 jobs:
+  check-commit-with-tag-and-old-tests:
+    runs-on: ubuntu-20.04
+    name: Check latest commit with tag and its checks
+    outputs:
+      NEED_TO_RUN_TESTS: ${{ steps.decision.outputs.NEED_TO_RUN_TESTS }}
+      COMMIT_WITH_TAG: ${{ steps.decision.outputs.COMMIT_WITH_TAG }}
+    steps:
+      - name: Git clone
+        uses: actions/checkout@v4
+      - name: Check latest commit with tag
+        id: latest_commit_with_test
+        run: |
+          git fetch origin $GITHUB_BASE_REF
+          git fetch origin rc --tags
+          diff=$(git log origin/$GITHUB_BASE_REF..origin/rc --oneline --format="%H")
+          echo "diff=$diff"
+          latestCommitWithTag=''
+          for commit in $diff; do
+            echo "checking commit $commit for tag"
+            tag=$(git tag --points-at "$commit")
+            if [[ -n "$tag" ]]; then
+              latestCommitWithTag=$commit
+              break
+            fi
+          done
+          echo "latestCommitWithTag=$latestCommitWithTag"
+          echo "latestCommitWithTag=$latestCommitWithTag" >> $GITHUB_OUTPUT
+
+      - name: Fetch checks from GitHub
+        id: response_from_github
+        if: steps.latest_commit_with_test.outputs.latestCommitWithTag != ''
+        run: |
+          latestCommitWithTag=${{ steps.latest_commit_with_test.outputs.latestCommitWithTag }}
+          response=$(curl \
+            -X GET \
+            -H "Accept: application/vnd.github+json" \
+            -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" \
+            -H "X-GitHub-Api-Version: 2022-11-28" \
+            https://api.github.com/repos/fingerprintjs/fingerprint-pro-cloudfront-integration/commits/$latestCommitWithTag/check-runs)
```

Comment:
As I remember, you could also use [cli app](https://cli.github.com/manual/gh_api) to do same things

Reply:
Updated